src/models/lazy_load.rs

```diff
@@ -807,3 +821,47 @@ impl<T: Clone + 'static> LazyItemVec<T> {
         items.transactional_update(|_| Vec::new()).unwrap();
     }
 }
+
+impl<T: Clone + 'static, const N: usize> LazyItemArray<T, N> {
+    pub fn new() -> Self {
+        let arr: [Option<LazyItem<T>>; N] = std::array::from_fn(|_| None);
+        Self {
+            items: STM::new(arr, 1, true),
+        }
+    }
+
+    pub fn insert(&self, index: usize, value: LazyItem<T>) {
+        let mut arc = self.items.clone();
+
+        arc.transactional_update(|set| {
+            let mut arr = set.clone();
+            arr[index] = Some(value.clone());
+            arr
+        })
+        .unwrap();
+    }
+
+    pub fn checked_insert(&self, index: usize, value: LazyItem<T>) -> Option<LazyItem<T>> {
+        let mut arc = self.items.clone();
+
+        let (_, new_child) = arc.arcshift.rcu_project(
+            |arr| {
+                (arr.get(index).unwrap().is_none())
+                    .then(|| {
+                        let mut new_arr = arr.clone();
+                        new_arr[index] = Some(value.clone());
+                        Some(new_arr)
+                    })
+                    .flatten()
+            },
+            |item| item.get(index).map(|item| item.clone()),
+        );
+
+        new_child.flatten()
+    }
+
+    pub fn get(&self, index: usize) -> Option<LazyItem<T>> {
+        let mut arc = self.items.clone();
```

Comment:
Is it possible to avoid a clone for a get here?

Reply:
Removed clone